pkg/controller/servicemesh/webhooks/validation/memberroll.go

```diff
@@ -185,14 +190,14 @@ func (v *MemberRollValidator) isUserAllowedToUpdatePodsInAllNamespaces(ctx conte
 	var wg sync.WaitGroup
 	wg.Add(numConcurrentSARChecks)
 	for i := 0; i < numConcurrentSARChecks; i++ {
-		go func() {
-			workerCtx := common.NewContextWithLogValues(ctx, "worker", i)
+		go func(workerNumber int) {
+			workerCtx := common.NewContextWithLogValues(ctx, "worker", workerNumber)
 			for ns := range in {
 				allowed, err := v.isUserAllowedToUpdatePods(common.NewContextWithLogValues(workerCtx, "namespace", ns), req, ns)
 				out <- result{namespace: ns, allowed: allowed, err: err}
 			}
 			wg.Done()
-		}()
+		}(i)
```

Comment:
interesting... I wonder what kind of obscure bugs could have happened here

Reply:
Fortunately, this only caused the wrong worker number to be printed in the logs.